README-FIPS.md

```diff
@@ -1,4 +1,445 @@
+also activates the "base" provider. The base provider does not include any
+cryptographic algorithms (and therefore does not impact the validation status of
+any cryptographic operations), but does include other supporting algorithms that
+may be required. It is designed to be used in conjunction with the FIPS module.
+
+This approach has the primary advantage that it is simple, and no code changes
+are required in applications in order to benefit from the FIPS module. There are
+some disadvantages to this approach:
+
+- You may not want all applications to use the FIPS module. It may be the case
+that some applications should and some should not.
+- If applications take explicit steps to not load the default config file or set
+different settings then this method will not work for them
+- The algorithms available in the FIPS module are a subset of the algorithms
+that are available in the default OpenSSL Provider. If those applications
+attempt to use any algorithms that are not present, then they will fail.
+- Usage of certain deprecated APIs avoids the use of the FIPS module. If any
+applications use those APIs then the FIPS module will not be used.
+
+Selectively making applications use the FIPS module by default
+--------------------------------------------------------------
+
+A variation on the above approach is to do the same thing on an individual
+application basis. The default OpenSSL config file depends on the compiled in
+value for OPENSSLDIR as described in the section above. However it is also
+possible to override the config file to be used via the `OPENSSL_CONF`
+environment variable. For example the following on Unix will cause the
+application to be executed with a non-standard config file location:
+
+    $ OPENSSL_CONF=/my/non-default/openssl.cnf myapplication
+
+Using this mechanism you can control which config file is loaded (and hence
+whether the FIPS module is loaded) on an application by application basis.
+
+This removes the disadvantage listed above that you may not want all
+applications to use the FIPS module. All the other advantages and disadvantages
+still apply.
+
+Programmatically loading the FIPS module (default library context)
+------------------------------------------------------------------
+
+Applications may choose to load the FIPS provider explicitly rather than relying
+on config to do this. The config file is still necessary in order to hold the
+FIPS module config data (such as its self test status and integrity data). But
+in this case we do not automatically activate the FIPS provider via that config
+file.
+
+To do things this way configure as per the section "Making all applications use
+the FIPS module by default" above, but edit the `fipsmodule.cnf` file to remove
+or comment out the line which says `activate = 1` (note that setting this value
+to 0 is **not** sufficient). This means all the required config information will
```

Comment:
Should we actually change the code that parses the activate value to allow setting 0 here to have the expected meaning?

Reply:
I seem to recall there might have been some difficulties with doing so. @paulidale might be able to comment.